daemon/src/main/java/org/lsposed/lspd/cli/Main.java

```diff
@@ -0,0 +1,151 @@
+import android.os.ServiceManager;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.ParameterException;
+
+import org.lsposed.lspd.ILSPManagerService;
+import org.lsposed.lspd.cli.exception.FormatException;
+import org.lsposed.lspd.cli.exception.NotXposedModuleException;
+import org.lsposed.lspd.cli.exception.RunException;
+import org.lsposed.lspd.cli.exception.UninstalledPackageException;
+import org.lsposed.lspd.cli.exception.UnknownCommandException;
+import org.lsposed.lspd.cli.handler.ICommand;
+import org.lsposed.lspd.cli.handler.LogHandler;
+import org.lsposed.lspd.cli.handler.ModulesHandler;
+import org.lsposed.lspd.cli.handler.ScopeHandler;
+import org.lsposed.lspd.cli.handler.StatusHandler;
+import org.lsposed.lspd.service.ILSPApplicationService;
+import org.lsposed.lspd.util.SignInfo;
+
+import static org.lsposed.lspd.cli.Utils.CMDNAME;
+import static org.lsposed.lspd.cli.Utils.ERRCODES;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class Main {
+
+    private static ILSPManagerService getLSPManagerService() throws RemoteException {
+        var activity_service = ServiceManager.getService("activity");
+        var heartBeat = new Binder();
+        Parcel data = Parcel.obtain();
+        Parcel reply = Parcel.obtain();
+        data.writeInterfaceToken("LSPosed");
+        data.writeInt(2);
+        data.writeString("lsp-cli:" + SignInfo.CLI_UUID);
+        data.writeStrongBinder(heartBeat);
+
+        var resArr = new ArrayList<IBinder>(1);
+        if(activity_service.transact(1598837584, data, reply, 0)) {
+            reply.readException();
+            var serviceBinder = reply.readStrongBinder();
+            if (serviceBinder == null) {
+                System.err.println("binder null");
+                return null;
+            }
+            var service = ILSPApplicationService.Stub.asInterface(serviceBinder);
+            if(service.requestInjectedManagerBinder(resArr) == null) {
+                System.err.println("not a manager");
+                return null;
+            }
+            if (resArr.size() > 0) {
+                return ILSPManagerService.Stub.asInterface(resArr.get(0));
+            } else {
+                System.err.println("Error: request injected return is empty");
+            }
+        } else {
+            System.err.println("Transact fail");
+        }
+
+        return null;
+    }
+
+    private static int exec(String[] args) {
+        LogHandler objLogHand = new LogHandler();
+        ModulesHandler objModHand = new ModulesHandler();
+        ScopeHandler objScopeHand = new ScopeHandler();
+        StatusHandler objStatusHand = new StatusHandler();
+
+        // TODO evaluate if generate the list at runtime, so just add a class and we have a new subcommand
+        var jc = JCommander.newBuilder()
+                     .programName(CMDNAME)
+                     .addCommand(objLogHand)
```

Comment:
Why not `addCommand("log", logCommand)`

Reply:
Because if we want change the name, there is only one place in the code where change it.
In original code was `addCommand("log", objLogHand)`, but it's redundant there is parameter in class LogHandler which define the command name. 
If you like with string name, I can restore the previous code.

Reply:
with refactor should resolved